google-cloud-pubsub/src/test/java/com/google/cloud/pubsub/spi/v1/FakeScheduledExecutorService.java

```diff
@@ -182,6 +204,13 @@ public void execute(Runnable command) {
       pendingCallables.add(callable);
     }
     work();
+    synchronized (expectedWorkQueue) {
+      if (!expectedWorkQueue.isEmpty() && expectedWorkQueue.peek().equals(callable.delay)) {
```

Comment:
Add a note that we are comparing by the delay in order to not depend as much on internal details.